### PR TITLE
HelpCenter: add warning in contact page when user has an active ticket

### LINKED ISSUE #64983: Help Center: Banner if open tickets
### What

Discussion here: pdm0RZ-bG-p2#comment-430

Add a banner to the contact page like:
image

if a user has open tickets with us, as it has proven to be an effective method of deflecting duplicate tickets

Can be used as a last check before sending the email/opening chat. `Hey, we see you have an open ticket created x minutes ago. Are you sure you want to open a new one?` We get the status of the ticket, when it was created, etc. so we can be quite smart about the message here if we want.

### PR DESCRIPTION
**Pull request by heavyweight**
#### Proposed Changes

* This PR adds a check if the current user has an active support ticket open. If this is the case we display a warning banner on the help center contact page.

| Before  | After |
| ------------- | ------------- |
| Screenshot 2022-07-13 at 16 45 39  | Screenshot 2022-07-13 at 16 44 44 |

#### Testing Instructions
- Use this diff D86223-code
 - Checkout this branch
 - In apps/editing-toolkit run `yarn dev --sync`
 - In a sandboxed simple site open the new help center
 - Make sure you have active support ticket, if not open one for testing
 - Click on `Still need help?` and check to see if the banner is displayed.
 - Make sure the banner is not displayed when there are no active support tickets.
 
**Note**
If you don't have an open support ticket, and you open one for testing purposes be patient.
It can take a while for the ticket to be created and returned with the `support-history` response.

#### Pre-merge Checklist

Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.

- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in [Simple](P9HQHe-k8-p2-p2), [Atomic](P9HQHe-jW-p2-p2), and [self-hosted Jetpack sites](PCYsg-g6b-p2-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Are we memoizing when appropriate (for expensive computations)? More info in [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md) and [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors)
- [ ] Have we sent any new strings [for translation](PCYsg-1vr-p2-p2) ASAP?

Related to #64983

### CODE REVIEW

File: packages/help-center/src/hooks/index.ts [hunk 1]
```diff
@@ -0,0 +1,4 @@
+export { useActiveSupportTicketsQuery } from './use-active-support-tickets-query';
```
Comment (heavyweight) [inline packages/help-center/src/hooks/index.ts]: Is it ok to have one index file for the hooks?
It made sense for me..

File: packages/help-center/src/hooks/use-active-support-tickets-query.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
+/* eslint-disable no-restricted-imports */
```
Comment (heavyweight) [inline packages/help-center/src/hooks/use-active-support-tickets-query.ts]: This was extracted [from here](https://github.com/Automattic/wp-calypso/blob/trunk/client/data/help/use-active-support-tickets-query.js)
I was undecided on whether or not to move it to `@automattic/data-stores`, and it ended up here.
Any strong opinions?

File: packages/help-center/src/components/help-center-notice.tsx [hunk 3]
```diff
@@ -63,12 +65,44 @@ export function HelpCenterOwnershipNotice( {
...
+	}
+
+	return (
+		<HelpCenterNotice>
+			<p>
+				<strong>
+					{ sprintf(
+						/* translators: %s humanized date ex: 2 hours ago */
+						__( 'You submitted a request %s.' ),
+						tickets[ 0 ].when,
```
Comment (heavyweight) [inline packages/help-center/src/components/help-center-notice.tsx]: From what I see `when` should be translated at the backend in support-history.php?r=19d11c4d#72
So this means no need for frontend translation of it, but when I test with a different language is not translating properly. Any ideas?

File: packages/help-center/src/hooks/index.ts (hunk 1 above)
Comment (heavyweight) [reply packages/help-center/src/hooks/index.ts]: will remove it since we don't do it

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7457666

Thank you @heavyweight for including a screenshot in the description! This is really helpful for our translators.

File: packages/help-center/src/hooks/index.ts (hunk 1 above)
Comment (alshakero) [reply packages/help-center/src/hooks/index.ts]: I suggest that we don't, to make tree shaking easier for Webpack and deleting unused code easier for us in the future.

File: packages/help-center/src/hooks/use-active-support-tickets-query.ts (hunk 2 above)
Comment (alshakero) [reply packages/help-center/src/hooks/use-active-support-tickets-query.ts]: Not really. We can move it if we decide we need to.

File: packages/help-center/src/components/help-center-notice.tsx [hunk 4]
```diff
@@ -63,12 +64,43 @@ export function HelpCenterOwnershipNotice( {
...
+		return null;
+	}
+
+	return (
+		<HelpCenterNotice>
+			<p>
+				<strong>
+					{ sprintf(
+						/* translators: %s humanized date ex: 2 hours ago */
+						__( 'You submitted a request %s.' ),
```
Comment (escapemanuele) [inline packages/help-center/src/components/help-center-notice.tsx]: We should use the domain, like `__( 'You submitted a request %s.', __i18n_text_domain__)`

File: packages/help-center/src/components/help-center-notice.tsx [hunk 5]
```diff
@@ -63,12 +64,43 @@ export function HelpCenterOwnershipNotice( {
...
+						tickets[ 0 ].when
+					) }
+				</strong>
+				&nbsp;
+				{ __( `Rest assured that we got your message and we'll be in touch as soon as we can.` ) }
```
Comment (escapemanuele) [inline packages/help-center/src/components/help-center-notice.tsx]: Also here we need the domain

**Review by escapemanuele — CHANGES_REQUESTED** [review]
Well done 👍 💯 ! I left a couple of minor comments.
Regarding the humanized date, it's better to ask for the best way of solving this to
in the translators channel 👍 

Anyway, the strings we introduced here are not translated yet, so you could ship this only to `en` locale, or add the label `String Freeze` to this PR, that should trigger translations

File: packages/data-stores/src/support-queries/use-active-support-tickets-query.ts [hunk 6]
```diff
@@ -0,0 +1,32 @@
+import { useQuery } from 'react-query';
+import wpcomRequest from 'wpcom-proxy-request';
+import { SupportTicket } from './types';
+
+const ACTIVE_STATUSES = [ 'New', 'Open', 'Hold' ];
+
+export const useActiveSupportTicketsQuery = ( email: string, queryOptions = {} ) => {
```
Comment (alshakero) [inline packages/data-stores/src/support-queries/use-active-support-tickets-query.ts]: In general, when migrating code from Calypso, please try to use the moved code in Calypso and not duplicate it. I mean, this [hook](https://github.com/Automattic/wp-calypso/blob/trunk/client/data/help/use-active-support-tickets-query.js#L6). The duplicate in Calypso should be deleted and the one in the package should be the only copy.

If TypeScript gives you weird errors, it likely means there is a circular dependency (Calypso depends on help-center and help-center on Calypso). So if that happens, this hook need to be in data-stores and both Calypso and HC can use it.

**Review by alshakero — CHANGES_REQUESTED** [review]
Awesome work. All works as expected. But the `ago` issue is a backend issue and I tried sending the `locale` param to no avail. Consider moving [this function](https://github.com/Automattic/wp-calypso/blob/trunk/apps/notifications/src/panel/templates/block-user.jsx#L20-L60) to a package (maybe `i18n-utils`) and use it to work out the relative time on the client side and be happy.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7457666).

**Review by alshakero — APPROVED** [review]
Optional but important for the future: it's highly encouraged to move code to `packages`, copying the`useActiveSupportTicketsQuery` hook to a package was great! I just meant please use the package version in Calypso too, not keep two copies. I see that you reverted to using the Calypso version in both places, my bad that I wasn't clear. 

The current solution works for sure, but the more you lean away from Calypso, the easier it is to manage our dependencies, it's hard to control what a Calypso import pulls with it.

File: apps/notifications/package.json [hunk 7]
```diff
@@ -26,6 +26,7 @@
 	"dependencies": {
 		"@automattic/calypso-color-schemes": "workspace:^",
 		"@automattic/calypso-polyfills": "workspace:^",
+		"@automattic/i18n-utils": "workspace:^",
```
Comment (alshakero) [inline apps/notifications/package.json:29]: If you want to test this:

1. cd `apps/notifications`.
2. run `yarn dev --sync`.
3. sandbox widgets.wp.com.
4. go to http://widgets.wp.com/notifications/

**Comment by heavyweight** [comment]
> Optional but important for the future: it's highly encouraged to move code to `packages`, copying the`useActiveSupportTicketsQuery` hook to a package was great! I just meant please use the package version in Calypso too, not keep two copies. I see that you reverted to using the Calypso version in both places, my bad that I wasn't clear.
> 
> The current solution works for sure, but the more you lean away from Calypso, the easier it is to manage our dependencies, it's hard to control what a Calypso import pulls with it.

I reverted this in an effort to reduce noise on the PR.
One thing that pushed me this way is I also saw us using `useHelpSearchQuery` from the same place.
I will open a follow-up PR extracting both.

**Review by alshakero — APPROVED** [review]
Perfect. No need for for the follow up PR. Let's only move code to packages as we need it.

**Pull request merged**